packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -197,6 +201,34 @@ export default createRule<Options, MessageId>({
       );
     }
 
+    function isMemberExpressionWithExactOptionalProperty(
+      node: TSESTree.MemberExpression,
+    ): boolean {
+      if (node.computed) {
+        // TODO - figure out how to handle when computed key used.
+        /*
+        declare const foo: { bar: { baz?: number }; };
+        declare const baz: 'baz';
+        foo.bar[baz] ??= 1;
+        */
```

Comment:
[Feature] Yeah we'll have to do something here to make this PR mergable.

Have you tried `getStaticValue`? That's kind of the standard utility we've been going with.

For _really_ complex cases where folks are doing funky stuff like `(1 * 1) + "other"` then it's fine to not handle them.

Reply:
Hi @JoshuaKGoldberg, While looking into handling the computed key case, I found another false positive which occurs regardless of the `exactOptionalPropertyTypes`. 
- [demo](https://typescript-eslint.io/play#ts=4.9.3&showAST=types&fileType=.tsx&code=KYOwrgtgBA0sCeBnKBvAUFKBBKBeKA5FgQDQZQBCehFpaAvmgC7wAOwUAYgPbfXqYA2nCQA6LAF0AXFHAQARsABOUAD6ywAG00BucsISJRFaRoXK99PQBNgAY00BDJRzvcQiJlABmvGT24beycXKDcPLwBrBBkRRD1fbkFo%2BAkoAH50-ABGPSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)
```ts
enum Keys {
  A = 'A',
  B = 'B',
}
type Foo = {
  [Keys.A]: number | null;
  [Keys.B]: number;
};
declare const foo: Foo;
declare const key: Keys;
foo[key] ??= 1; // Unnecessary conditional, expected left-hand side of `??` operator to be possibly null or undefined. 11:1 - 11:9

let bar = foo[key];
bar ??= 1;
```

I think this false positive and the #6635 can be handled together, without branching based on `exactOptionalPropertyTypes`.
I'd like to create an issue for the newly discovered false positive and come back to it after it's been triaged, what do you think?

Reply:
Yeah that makes sense to me. Nice find!